di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/handlers/AuthorizeHandler.java

```diff
@@ -237,22 +246,46 @@ private String getSharedAttributes(QueryParamsMap queryParamsMap) {
         Map<String, String> client = clientConfig.get(clientIdParam);
 
         if (client == null) {
-            return "Could not find client configuration details for: " + clientIdParam;
+            return "Error: Could not find client configuration details for: " + clientIdParam;
         }
 
-        String sharedAttributesJson = null;
+        String sharedAttributesJson;
         if (!Validator.isNullBlankOrEmpty(requestParam)) {
             try {
                 SignedJWT signedJWT = SignedJWT.parse(requestParam);
 
+                if (isInvalidSignature(signedJWT, client)) {
+                    LOGGER.error("JWT signature is invalid");
+                    return "Error: Signature of the shared attribute JWT is not valid";
+                }
+
                 JWTClaimsSet claimsSet = signedJWT.getJWTClaimsSet();
                 Map<String, Object> claimsMap = claimsSet.toJSONObject();
                 sharedAttributesJson = gson.toJson(claimsMap);
             } catch (ParseException e) {
                 LOGGER.error("Failed to parse the shared attributes JWT");
                 sharedAttributesJson = "Error: failed to parse shared attribute JWT";
+            } catch (JOSEException | CertificateException e) {
+                LOGGER.error("Failed to verify the signature of the JWT", e);
+                sharedAttributesJson = "Error: failed to verify the signature of the shared attribute JWT";
             }
+        } else {
+            sharedAttributesJson = "Error: missing 'request' query parameter";
         }
         return sharedAttributesJson;
     }
+
+    private boolean isInvalidSignature(SignedJWT signedJWT, Map<String, String> client)
+            throws CertificateException, JOSEException {
+        byte[] binaryCertificate =
+                Base64.getDecoder().decode(client.get(SIGNING_CERT_PARAM));
+        CertificateFactory factory = CertificateFactory.getInstance("X.509");
+        Certificate certificate = factory.generateCertificate(new ByteArrayInputStream(binaryCertificate));
+
+        PublicKey publicKey = certificate.getPublicKey();
```

Comment:
Just an observation: if we're just using the x509 cert as a way of storing the public key, can we just have the public key in config instead? If there's intent to validate the expiry and chain of the cert later on, then I guess this makes sense.

Reply:
We had a bit of a chat about this and decided on going down the X509 route. Mostly because it'll be consistent with what we're doing elsewhere, so in a few months we don't go "oh, why did we do this differently?".

Reply:
There's probably a discussion to have at some point about whether a "current set" of public keys could be loaded into configuration somewhere, so that we don't have to be parsing certificates all over the place on every request, but I'm all for not prematurely optimising.

eg we might well want a check for revocation when we check the cert/chain, but not necessarily on every single request.

(And cc @philandstuff as it's more his wheelhouse than mine!)